compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -7061,4 +7070,171 @@ private STNode parseTypeCastParam() {
 
         return STNodeFactory.createTypeCastParamNode(annot, type);
     }
+
+    /**
+     * Parse table constructor expression.
+     * <p>
+     * <code>
+     * table-constructor-expr := table [key-specifier] [ [row-list] ]
+     * </code>
+     *
+     * @return Parsed node
+     */
+    private STNode parseTableConstructorExpr() {
+        startContext(ParserRuleContext.TABLE_CONSTRUCTOR);
+        STNode tableKeyword = parseTableKeyword();
+        STToken nextToken = peek();
+
+        STNode keySpecifier;
+        if (nextToken.kind == SyntaxKind.KEY_KEYWORD) {
+            keySpecifier = parseKeySpecifier();
+        } else {
+            keySpecifier = STNodeFactory.createEmptyNode();
+        }
```

Comment:
Will this have a problem if there is some additional token before `key` keyword:
for eg: `table foo key() [{k1:v1}]`.

Reply:
I think we'll have to do something like:
```
switch(nextTokenKind)
    case KEY: 
        // parse key-specifier
    case OPEN_BRACKET: 
        // parse row-list 
    default: 
        // if non of the above, then recover and re-parse/return
```

Reply:
fixed and two cases added.